colossalai/shardformer/layer/parallel_module.py

```diff
@@ -54,11 +54,11 @@ def _save_to_state_dict(self, destination, prefix, keep_vars):
         """
         for name, param in self._parameters.items():
             if param is not None:
-                destination[prefix + name] = gather_distributed_param(param, keep_vars=keep_vars)
+                destination[prefix + name] = gather_distributed_param(param, keep_vars=keep_vars).data
 
         for name, buf in self._buffers.items():
             if buf is not None and name not in self._non_persistent_buffers_set:
-                destination[prefix + name] = buf if keep_vars else buf.detach()
+                destination[prefix + name] = buf.data if keep_vars else buf.detach().data
```

Comment:
It seems this is unnecessary? as buffer won't be dtensor

Reply:
updated.